src/util.js

```diff
@@ -25,6 +27,52 @@ var Util = {
         message: null
     },
 
+    escapeRegExp: function(string) {
+        return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+    },
+
+    /**
+     * Returns the indices of each occurrence of searchString within
+     * contentString ignoring occurrence within widget references.
+     */
+    getIndicesOf: function(contentString, searchString) {
+        if (searchString === "") {
+            return [];
+        }
+
+        var indices = [];
+        var searchStringLength = searchString.length;
+        var pieces = this.split(contentString, this.rWidgetSplit);
```

Comment:
We should probably use `Util` instead of `this`. Using `this` means that other people must always call this method using the `Util` class, and can't pull it off as a raw function (for example, `_.map(myArray, Util.getIndicesOf)` wouldn't work)